crates/sui-kvstore/src/client.rs

```diff
@@ -126,15 +126,24 @@ impl KVWriteClient for DynamoDbClient {
         if items.is_empty() {
             return Ok(());
         }
-        for chunk in items.chunks(25) {
-            self.dynamo_client
+        let mut queue: VecDeque<Vec<_>> = items.chunks(25).map(|ck| ck.to_vec()).collect();
+        while let Some(chunk) = queue.pop_front() {
+            let response = self
+                .dynamo_client
                 .batch_write_item()
                 .set_request_items(Some(HashMap::from([(
                     self.table_name.clone(),
                     chunk.to_vec(),
                 )])))
                 .send()
                 .await?;
+            if let Some(response) = response.unprocessed_items {
+                if let Some(unprocessed) = response.into_iter().next() {
+                    if !unprocessed.1.is_empty() {
+                        queue.push_back(unprocessed.1);
```

Comment:
not sure if we want to add backoff here

Reply:
Yeah backoff would be great, I wonder if there are settings in dynamodb client which let us configure retries and backoff

Reply:
FWIU there's not, it's indeed a little bit confusing to not at least return the error
the reason I'm slightly hesitant about adding backoff is because of the concurrency model in the KV backfiller
we spawn N concurrent actors to process checkpoints, but store one single progress int value in "meta store" for the job
so it's updated only once there's continuation of progress from multiple actors
thus one slow actor can have impact on that metric. I guess we can try it anyway and see if there's any issue with this and it needs to be adjusted